halo/evmstaking2/keeper/keeper_internal_test.go

```diff
@@ -90,48 +91,53 @@ func TestInsertAndDeleteEVMEvents(t *testing.T) {
 	}
 }
 
-func TestDeliverDelegate(t *testing.T) {
+func TestDelivery(t *testing.T) {
 	t.Parallel()
 
 	submissionDelay := int64(3)
 	ethClientMock, err := ethclient.NewEngineMock(
 		ethclient.WithPortalRegister(netconf.SimnetNetwork()),
 		ethclient.WithMockSelfDelegation(k1.GenPrivKey().PubKey(), 1),
+		ethclient.WithMockValidatorCreation(k1.GenPrivKey().PubKey()),
 	)
 	require.NoError(t, err)
 
 	msgServer := &msgServerMock{}
 
-	keeper, ctx := setupKeeper(t, submissionDelay, ethClientMock, authKeeperMock{}, bankKeeperMock{}, stakingKeeperMock{}, msgServer)
+	keeper, ctx := setupKeeper(t, submissionDelay, ethClientMock, authKeeperMock{}, bankKeeperMock{}, &stakingKeeperMock{}, msgServer)
 
 	var hash common.Hash
 	events, err := keeper.Prepare(ctx, hash)
 	require.NoError(t, err)
 
-	require.Len(t, events, 1)
+	require.Len(t, events, 2)
 
 	for _, event := range events {
 		err := keeper.Deliver(ctx, hash, event)
 		require.NoError(t, err)
 	}
 
-	// Make sure the event was persisted.
-	insertedID := uint64(1)
-	found, err := keeper.eventsTable.Has(ctx, insertedID)
-	require.NoError(t, err)
-	require.True(t, found)
+	// Make sure the events were persisted.
+	for id := uint64(1); id < 3; id++ {
+		found, err := keeper.eventsTable.Has(ctx, id)
+		require.NoError(t, err)
+		require.True(t, found)
```

Comment:
feels like a helper function would help here, since we do this many times in all the tests:

```
assertContains(t, keeper, id)
assertNotContains(t, keeper, id)
```

This helps make tests much more readable.